src/main/java/org/owasp/html/HtmlPolicyBuilder.java

```diff
@@ -714,6 +735,32 @@ public AttributeBuilder matching(
       });
     }
 
+	/**
+     * Restrict the values allowed by later {@code allow*} calls to those
+     * NOT matching the pattern. This is a convenience method, as inverting
+     * certain patterns can otherwise be quite complex.
+     */
+    public AttributeBuilder notMatching(final Pattern pattern) {
+      return matching(new AttributePolicy() {
+        public @Nullable String apply(String elementName, String attributeName, String value) {
+          return !pattern.matcher(value).matches() ? value : null;
+        }
+      });
+    }
+
+    /**
+     * Restrict the values allowed by later {@code allow*} calls to those
+     * that are not only whitespace or an empty string. Can be combined 
+	 * with calls to {@code matching} or {@code notMatching}.
```

Comment:
Please use `{@link #matching}`  instead of `{@code matching}`.  The latter allows the reader to click through and generates doc warnings should APIs change.

Please fix indentation.

Maybe add a caveat along the lines of

> Sometimes empty attributes are different from absent attributes.  For example `<input checked="">` is not equivalent to `<input>` and `<a href="">` is a link to the current document without a fragment while `<a>` is not.